jama5262/Politico
# #164057331 Finished with client and db custom error messages

**What does this PR do?**
Sets the custom error message validation for client and database errors

**Description of Task to be completed?**

1. Set all validation for client
2. Set custom messages validation for database errors

**How should this be manually tested?**
Go to [Github Page](https://github.com/jama5262/Politico), make sure you are on the develop branch

1. Make sure to have the python virtualenv if not `pip install virtualenv`
2. Create the env with this command `virtualenv venv`
3. For windows `venv\Scripts\activate`, for Mac please Google to activate the env
4. Install the packages from the requirements.txt by using this commands `pip install -r requirements.txt`
5. Open POSTMAN to test the following endpoints
6. To test the validations try adding the following data to any endpoint
6.1. E.g POST parties endpoint `https://politico-andela-37.herokuapp.com/api/v2/parties' adding unneeded properties
```
{
    "name": " Party Name",
    "abbr": "PN",
    "logo_url": "Party URL",
    "hq_address": "PartyQ",
    "UNEEDED_PROPERTY" : "SOME DATA"
}
```
The following validation error should be displayed
```
{
    "error": "422 (Unprocessable Entity), Please make sure to enter the correct requests, which are ('name', 'hq_address', 'logo_url', 'abbr')",
    "status": 422
}
```
6.2 E.g POST parties endpoint `https://politico-andela-37.herokuapp.com/api/v2/parties' adding white space
```
{
    "name": "    ",
    "abbr": "JP",
    "logo_url": "   ",
    "hq_address": "PartyQ"
}
```
The following validation error should be displayed
```
{
    "error": "422 (Unprocessable Entity), Please make sure the values do not have whitespaces",
    "status": 422
}
```
6.3 E.g POST parties endpoint `https://politico-andela-37.herokuapp.com/api/v2/parties' uncapitalized 
values
```
{
    "name": "party name",
    "abbr": "JP",
    "logo_url": "Party URL",
    "hq_address": "PartyQ"
}
```
Will automatically corrected
```
{
     "data": {
        "abbr": "JP",
        "hq_address": "PartyQ",
        "logo_url": "Party URL",
        "name": "Party Name"
    },
    "status": 200
}
```
7 Database errors
7.1 E.g POST parties endpoint `https://politico-andela-37.herokuapp.com/api/v2/parties'  entering same name
```
{
    "name": "party name",
    "abbr": "JP",
    "logo_url": "Party URL",
    "hq_address": "PartyQ"
}
```
The following validation error should be displayed
```
{
    "error": "Error, name=Party 1 already exists.",
    "status": 500
}
```

**What are the relevant pivotal tracker stories?**
[#164057331](https://www.pivotaltracker.com/n/projects/2241896)

**Screenshots**
![capture](https://user-images.githubusercontent.com/18502400/53014669-2a20bb80-345a-11e9-9e22-a10c2f45ad28.PNG)
![capture](https://user-images.githubusercontent.com/18502400/53014715-54727900-345a-11e9-80e8-fa8d740d471d.PNG)
![capture](https://user-images.githubusercontent.com/18502400/53014778-84ba1780-345a-11e9-8844-e208a86c7feb.PNG)
